### PR TITLE
PLT-6169: Batch add users to teams.

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
Batch Add users to teams.

Having some trouble with tests passing in isolation but not passing when run together. Pushing anyway, as Jenkins might not have the same problem, and I don't want to delay code review on this any longer.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-6169

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (required for all new APIs) https://github.com/mattermost/mattermost-api-reference/pull/201
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

File: api4/team.go [hunk 1]
```diff
@@ -340,6 +345,51 @@ func addTeamMember(c *Context, w http.ResponseWriter, r *http.Request) {
...
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+
+	var err *model.AppError
+	members := model.TeamMembersFromJson(r.Body)
+
+	if (len(members) > MAX_ADD_MEMBERS_BATCH || len(members) == 0) {
+		c.SetInvalidParam("")
```
Comment (jwilander) [inline api4/team.go]: Maybe just specify something in the `SetInvalidParam` so the requester knows what field is in correct

File: api4/team.go [hunk 2]
```diff
@@ -340,6 +345,51 @@ func addTeamMember(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	var userIds []string
+	for _, member := range members {
+		if member.TeamId != c.Params.TeamId {
+			c.SetInvalidParam("")
```
Comment (jwilander) [inline api4/team.go]: Same for here, maybe specify `team_id for member with user_id={user_id}`

**Comment by grundleborg** [comment]
@jwilander fixed.

File: api4/team.go [hunk 3]
```diff
@@ -340,6 +345,51 @@ func addTeamMember(c *Context, w http.ResponseWriter, r *http.Request) {
...
+
+	if len(members) > MAX_ADD_MEMBERS_BATCH || len(members) == 0 {
+		c.SetInvalidParam("too many members in batch")
+		return
+	}
+
+	var userIds []string
+	for _, member := range members {
+		if member.TeamId != c.Params.TeamId {
+			c.SetInvalidParam("team_id for member with user_id=" + member.UserId)
```
Comment (hmhealey) [inline api4/team.go]: Should these messages be localized since they're not just the name of the invalid argument?

Comment (jwilander) [inline api4/team.go]: I would lean towards no since these are for developers not users

Comment (enahum) [inline api4/team.go]: yeah I don't think they need to be localized

**Pull request merged**